pkg/controller/controller_ref_manager.go

```diff
@@ -349,7 +349,7 @@ func (m *ReplicaSetControllerRefManager) AdoptReplicaSet(rs *extensions.ReplicaS
 	// Note that ValidateOwnerReferences() will reject this patch if another
 	// OwnerReference exists with controller=true.
 	addControllerPatch := fmt.Sprintf(
-		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
+		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s","finalizers": ["foregroundDeletion"]}}`,
```

Comment:
Can you add a unit test to test the patch? Something similar to https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/garbagecollector_test.go#L538-L588.